harness/features/variable.local.test.ts

```diff
@@ -0,0 +1,193 @@
+                    && body.message.includes('onClientInitialized was invoked')) resolveIsInitialized(true)
+                return [200]
+            })
+
+            scope.persist().post(`/client/${clientId}/v1/events/batch`).reply(function(uri, body) {
+                mockEvents(body)
+                return [200]
+            })
+        })
+
+        afterEach(() => {
+            jest.clearAllMocks()
+        })
+
+        describeIf(capabilities.includes(Capabilities.local))(name, () => {
+            describe('uninitialized client', () => {
+                it('should return default value if client is uninitialized',  async () => {
+                    const variableResponse = await callVariable(
+                        clientId, url, variationOnUser, 'string-var', 'string default'
+                    )
+                    const variable = await variableResponse.json()
+                    expectDefaultValue(variable, 'string default')
+                })
+
+                it('should return default value if user is invalid',  async () => {    
+                    const variableResponse =
+                        await callVariable( clientId, url, invalidUser, 'string-var', 'string default')
+                    const variable = await variableResponse.json()
+                    expectDefaultValue(variable, 'string default')
+                })
+            })
+
+            describe('initialized client', () => {
+                beforeAll(async () => {
+                    const cbResponse = await callOnClientInitialized(clientId, url, baseURL)
+                    await cbResponse.json()
+
+                    await isInitialized
+                })
+
+                it('should return variable if mock server returns object matching default type',  async () => {
+                    const variableResponse = await callVariable(
+                        clientId, url, variationOnUser, 'string-var', 'default_val'
+                    )
+                    const variable = await variableResponse.json()
+                    expect(variable.entityType).toBe('Variable')
+                    expect(variable.data.value).toBe('string')
+                    expect(variable.data.isDefaulted).toBe(false)
+                    expect(variable.data.defaultValue).toBe('default_val')
+
+                    await expectEventSent('user1', 'string-var', 'aggVariableEvaluated')
+                })
+
+                it('should throw exception if user is invalid',  async () => {    
+                    const variableResponse =
+                        await callVariable( clientId, url, invalidUser, 'string-var', 'string default')
+                    const variable = await variableResponse.json()
+
+                    expect(variable.exception).toBe('Must have a user_id set on the user')
+                })
+
+                it('should return default value if default type doesn\'t match variable type',  async () => {    
```

Comment:
i have made a helper method to be able to call a suite of tests with the different types of variables, so you can put the tests needing to test variable values / default values in one describe and use the helper method i made, `forEachVariableType`

Reply:
added both!